scanpy_scripts/lib/_filter.py

```diff
@@ -158,11 +180,14 @@ def _get_attributes(adata):
 
 def _attributes_exists(name, attributes, dtype):
     cond_cat = ''
+
     if name.startswith('c:') or name.startswith('g:'):
         cond_cat, _, cond_name = name.partition(':')
-        found = int(cond_name in attributes[cond_cat][dtype])
+        test_name = cond_name[1:] if cond_name.startswith('!') else cond_name
```

Comment:
This was because for Boolean filters I was just passing the variable name- and I wanted to be able to negate it. Obviously not relevant if you don't fancy Boolean filters.

Reply:
Under the changes introduced before scanpy-1.4.2/anndata-0.6.19, boolean attributes are stored as categorical. If 'mito' is already in the object before filtering, it should be a categorical. The only exception is when it's generated during the filtering, i.e.
https://github.com/ebi-gene-expression-group/scanpy-scripts/blob/6313d73820cde8fc3a4fcba07d9a0fecc19ff555/scanpy_scripts/lib/_filter.py#L38
I therefore think it's much simpler converting it to categorical after that line.